pallets/collator-allowlist/src/lib.rs

```diff
@@ -47,7 +49,10 @@ pub mod pallet {
 		type WeightInfo: WeightInfo;
 
 		/// The Validator Id type
-		type ValidatorId: Member + Parameter + MaybeSerializeDeserialize;
+		type ValidatorId: Member
+			+ Parameter
+			+ MaybeSerializeDeserialize
+			+ sp_std::convert::TryFrom<Self::AccountId>;
```

Comment:
what does ValidatorId repesent here ?
Why can't we use `IsType` if the they are of the same types?

Reply:
`ValidatorId` is any type that implements those traits. I guess we _could_ do that, but we wouldn't want our design to be unnecessarily restrictive but instead follow what the pallet_session, which we sort of extend, does: https://paritytech.github.io/substrate/master/src/pallet_session/lib.rs.html#380-384

Reply:
Not sure I like this new type :-D
Since ValidatorRegistration has parametric type, we can directly set T::AccountId since the I see no use for having a seperate associate type.

I would like to move away from `TryFrom` since this introduces another Optional success which would ideally not be present since the types are same. If we want to follow pallet_session, then we can move to having a conversion type like they have it `ValidatorIdOf` so that mapping would be easier if the types are not same

Reply:
By directly setting `T::AccountId` as the type parameter of `ValidatorRegistration`, we make it impossible for this pallet to be used in runtimes where those two types are not the same. I will see what I can do 👍

Reply:
I dont like the tryFrom there but I'll leave this to you.

Reply:
I think we don't need that specific trait so we may be able do drop it, just verifying that 👍